src/savedata.c

```diff
@@ -845,585 +838,558 @@ int SaveTableEntry_BodySize (int saveTableID)
     return size;
 }
 
-static void sub_020251DC (SavePageInfo * param0)
+static void SavePageInfo_Init (SavePageInfo *pageInfo)
 {
-    const SaveTableEntry * v0 = gSaveTable;
-    int v1;
-    int v2 = 0;
+    const SaveTableEntry *saveTable = gSaveTable;
+    int i;
+    int totalSize = 0;
 
-    GF_ASSERT(gSaveTableSize == 38);
+    GF_ASSERT(gSaveTableSize == SAVE_TABLE_ENTRY_MAX);
 
-    for (v1 = 0; v1 < gSaveTableSize; v1++) {
-        GF_ASSERT(v0[v1].dataID == v1);
+    for (i = 0; i < gSaveTableSize; i++) {
+        GF_ASSERT(saveTable[i].dataID == i);
 
-        param0[v1].pageID = v0[v1].dataID;
-        param0[v1].size = SaveTableEntry_BodySize(v1);
-        param0[v1].location = v2;
-        param0[v1].checksum = 0;
-        param0[v1].blockID = v0[v1].blockID;
+        pageInfo[i].pageID = saveTable[i].dataID;
+        pageInfo[i].size = SaveTableEntry_BodySize(i);
+        pageInfo[i].location = totalSize;
+        pageInfo[i].checksum = 0;
+        pageInfo[i].blockID = saveTable[i].blockID;
 
-        v2 += param0[v1].size;
+        totalSize += pageInfo[i].size;
 
-        if ((v1 == gSaveTableSize - 1) || (v0[v1].blockID != v0[v1 + 1].blockID)) {
-            v2 += sizeof(SaveBlockFooter);
+        if ((i == gSaveTableSize - 1) || (saveTable[i].blockID != saveTable[i + 1].blockID)) {
+            totalSize += sizeof(SaveBlockFooter);
         }
     }
 
-    if (v1 % 4 != 0) {
-    }
-
-    GF_ASSERT(v2 <= 0x1000 * 32);
+    GF_ASSERT(totalSize <= SAVE_SECTOR_SIZE * SAVE_PAGE_MAX);
 }
 
-static void sub_02025258 (SaveBlockInfo * param0, const SavePageInfo * param1)
+static void SaveBlockInfo_Init (SaveBlockInfo *blockInfo, const SavePageInfo *pageInfo)
 {
-    int v0 = 0;
-    int v1, v2;
-    int v3, v4;
+    int totalSectors = 0, blockOffset = 0, block, page = 0;
 
-    v1 = 0;
-    v2 = 0;
-    v4 = 0;
+    for (block = 0; block < SAVE_BLOCK_ID_MAX; block++) {
+        blockInfo[block].saveBlockID = block;
+        blockInfo[block].size = 0;
 
-    for (v3 = 0; v3 < 2; v3++) {
-        param0[v3].saveBlockID = v3;
-        param0[v3].size = 0;
-
-        for (; param1[v4].blockID == v3 && v4 < gSaveTableSize; v4++) {
-            param0[v3].size += param1[v4].size;
+        for (; pageInfo[page].blockID == block && page < gSaveTableSize; page++) {
+            blockInfo[block].size += pageInfo[page].size;
         }
```

Comment:
I wonder if this is more easily grokked as a `do-while`:

```suggestion
        do {
            blockInfo[block].size += pageInfo[page].size;
            page++;
        } while (pageInfo[page].blockID == block && page < gSaveTableSize);
```